astropy/table/table.py

```diff
@@ -2533,6 +2534,9 @@ def write(self, *args, **kwargs):
         The arguments and keywords (other than ``format``) provided to this function are
         passed through to the underlying data reader (e.g. `~astropy.io.ascii.write`).
         """
+        if((ext=='.h5'or ext=='.hdf5')and kwargs['overwrite']):
```

Comment:
i actually forgot to add this---name, ext = os.path.splitext(args[0])--- before line 2537